zerver/tests/test_signup.py

```diff
@@ -462,7 +462,7 @@ def test_register(self) -> None:
         with queries_captured() as queries:
             self.register(self.nonreg_email('test'), "test")
         # Ensure the number of queries we make is not O(streams)
-        self.assert_length(queries, 73)
+        self.assert_length(queries, 81)
```

Comment:
This big increase probably suggests a bug in your changes where we're doing database queries in a loop.  We should figure out why and fix that.

Reply:
Do you understand what the 4 new queries here are?  The number still seems high.

Reply:
4 new queries are a result of the new implementation of `send_future_email` where we first create scheduled email then update it with associating users. This happens for 2 users hence 4 queries.

Reply:
I see -- it's a little silly that creating one of these went from 1 query to 3 in the common case (probably 99% of cases of sending emails in general) where there's only one recipient, to support the rare use case with multiple recipients.  Because there is a significant cost to adding 4 more queries to the new user creation process, it be worth the extra work of having both a nullable `user` field and ` nullable `users` field, and using the singleton `user` one when there's only one user.  What do you think about that approach?

Reply:
Sounds right to me.